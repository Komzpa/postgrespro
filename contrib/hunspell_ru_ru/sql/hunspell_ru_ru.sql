CREATE EXTENSION hunspell_ru_ru;

CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('распространённости'), ('межнационального'),
						('включается'), ('значу'), ('зевнув'),
						('зевоты'), ('землей'), ('волчатами'), ('мойтесь');

SELECT ts_lexize('public.russian_hunspell', name) FROM table1;

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('public.russian', "name"));
SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.russian', t.name) AS d;

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('public.russian', "name"));
SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.russian', t.name) AS d;