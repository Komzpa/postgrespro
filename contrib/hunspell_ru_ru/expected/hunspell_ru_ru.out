CREATE EXTENSION hunspell_ru_ru;
CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('распространённости'), ('межнационального'),
						('включается'), ('значу'), ('зевнув'),
						('зевоты'), ('землей'), ('волчатами'), ('мойтесь');
SELECT ts_lexize('public.russian_hunspell', name) FROM table1;
      ts_lexize       
----------------------
 {распространённость}
 {межнациональный}
 {включаться}
 {значить}
 {зевнуть}
 {зевота}
 {земля}
 {волчонок}
 {мыться}
(9 rows)

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('public.russian', "name"));
SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.russian', t.name) AS d;
 alias |    description    |       token        |          dictionaries           |    dictionary    |       lexemes        
-------+-------------------+--------------------+---------------------------------+------------------+----------------------
 word  | Word, all letters | распространённости | {russian_hunspell,russian_stem} | russian_hunspell | {распространённость}
 word  | Word, all letters | межнационального   | {russian_hunspell,russian_stem} | russian_hunspell | {межнациональный}
 word  | Word, all letters | включается         | {russian_hunspell,russian_stem} | russian_hunspell | {включаться}
 word  | Word, all letters | значу              | {russian_hunspell,russian_stem} | russian_hunspell | {значить}
 word  | Word, all letters | зевнув             | {russian_hunspell,russian_stem} | russian_hunspell | {зевнуть}
 word  | Word, all letters | зевоты             | {russian_hunspell,russian_stem} | russian_hunspell | {зевота}
 word  | Word, all letters | землей             | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | волчатами          | {russian_hunspell,russian_stem} | russian_hunspell | {волчонок}
 word  | Word, all letters | мойтесь            | {russian_hunspell,russian_stem} | russian_hunspell | {мыться}
(9 rows)

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('public.russian', "name"));
SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.russian', t.name) AS d;
 alias |    description    |       token        |          dictionaries           |    dictionary    |       lexemes        
-------+-------------------+--------------------+---------------------------------+------------------+----------------------
 word  | Word, all letters | распространённости | {russian_hunspell,russian_stem} | russian_hunspell | {распространённость}
 word  | Word, all letters | межнационального   | {russian_hunspell,russian_stem} | russian_hunspell | {межнациональный}
 word  | Word, all letters | включается         | {russian_hunspell,russian_stem} | russian_hunspell | {включаться}
 word  | Word, all letters | значу              | {russian_hunspell,russian_stem} | russian_hunspell | {значить}
 word  | Word, all letters | зевнув             | {russian_hunspell,russian_stem} | russian_hunspell | {зевнуть}
 word  | Word, all letters | зевоты             | {russian_hunspell,russian_stem} | russian_hunspell | {зевота}
 word  | Word, all letters | землей             | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | волчатами          | {russian_hunspell,russian_stem} | russian_hunspell | {волчонок}
 word  | Word, all letters | мойтесь            | {russian_hunspell,russian_stem} | russian_hunspell | {мыться}
(9 rows)

